cpp/open3d/data/ExtractZIP.cpp

```diff
@@ -0,0 +1,237 @@
+// ----------------------------------------------------------------------------
+// -                        Open3D: www.open3d.org                            -
+// ----------------------------------------------------------------------------
+// The MIT License (MIT)
+//
+// Copyright (c) 2018-2021 www.open3d.org
+//
+// Permission is hereby granted, free of charge, to any person obtaining a copy
+// of this software and associated documentation files (the "Software"), to deal
+// in the Software without restriction, including without limitation the rights
+// to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+// copies of the Software, and to permit persons to whom the Software is
+// furnished to do so, subject to the following conditions:
+//
+// The above copyright notice and this permission notice shall be included in
+// all copies or substantial portions of the Software.
+//
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+// IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+// AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING
+// FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS
+// IN THE SOFTWARE.
+// ----------------------------------------------------------------------------
+
+#include "open3d/data/ExtractZIP.h"
+
+// Reference:
+// https://github.com/madler/zlib/blob/master/contrib/minizip/miniunz.c
+
+#include <errno.h>
+#include <fcntl.h>
+#include <stdio.h>
+#include <stdlib.h>
+#include <string.h>
+#include <unzip.h>
+
+#include <iostream>
+
+#include "open3d/utility/FileSystem.h"
+#include "open3d/utility/Logging.h"
+
+#ifdef WIN32
+#include <direct.h>
+#include <io.h>
+#else
+#include <unistd.h>
+#endif
+
+#ifdef __APPLE__
+// In darwin and perhaps other BSD variants off_t is a 64 bit value, hence no
+// need for specific 64 bit functions
+#define FOPEN_FUNC(filename, mode) fopen(filename, mode)
+#else
+#define FOPEN_FUNC(filename, mode) fopen64(filename, mode)
+#endif
+
+#define CASESENSITIVITY (0)
+// If required in future, the `WRITEBUFFERSIZE` size can be increased to 16384.
+#define WRITEBUFFERSIZE (8192)
+#define MAXFILENAME (256)
+
+namespace open3d {
+namespace data {
+
+static int ExtractCurrentFile(unzFile uf, const std::string &password) {
```

Comment:
This is a copied C funtion, it is returning an int. We shall not throw an exception here. Let it return an integer and throw exception after that. 

Let's revert to the previous state.

Reply:
Done.

Reply:
Reverted the function to the previous state, with LogWarning, since it might be useful for debugging the error source in the future. The error is thrown in `ExtractAll` with the error code.